### PR TITLE
Deprecate View::set_global and ::bind_global [fixes #587]

### LINKED ISSUE #587: Deprecate View::set_global
In efforts to improve the architecture of Kohana, and move away from static global scope, I propose that we deprecate View::set_global and associated functionality in version 3.4.  PR upcoming if there is consensus.

### PR DESCRIPTION
**Pull request by acoulton**
These promote bad architectural decisions in end-user applications and
should be strongly discouraged.

### CODE REVIEW

**Comment by acoulton** [comment]
@enov can you review?

File: guide/kohana/mvc/views.md [hunk 1]
```diff
@@ -69,43 +69,40 @@ You can also assign variables directly to the View object.  This is identical to
 
 ### Global Variables
```
Comment (enov) [inline guide/kohana/mvc/views.md]: Maybe change the title to: "Dealing with global-like variables"

**Comment by acoulton** [comment]
@enov good catch, thanks.

**Comment by sergeyklay** [comment]
:+1:

**Pull request merged**